### PR TITLE
nginx: bump and add build variant and other fix

### PR DESCRIPTION
**Pull request by Ansuel**
This add 2 variant of nginx so we can have package with SSL config flag preselected. This also add support for 2 more module and upgrade gninx to latest version. Also add myself as secondary maintainer to apply luci modification quickly.
Use of autoreconf to fix problems with recompilation on every new build (even if the version is the same). Add a patch to ignore on invalid configure option instead of trow error.
@hnyman @heil 
Think the name for the build variant are not right... Any idea about that?

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

File: net/nginx/Makefile [hunk 1]
```diff
@@ -67,181 +67,235 @@ PKG_CONFIG_DEPENDS := \
...
 endef
 
 define Package/nginx/description
  nginx is an HTTP and reverse proxy server, as well as a mail proxy server,
  written by Igor Sysoev.
 endef
 
+define Package/nginx
+  $(Package/nginx/default)
+  VARIANT:=no-ssl
```
Comment (karlp) [inline net/nginx/Makefile]: iirc, if you want PROVIDES and DEPENDS to work properly, you can't have a variant with the name as the same virtual package

Comment (Ansuel) [reply net/nginx/Makefile]: what you mean with a variant with the name as the same virtual package

Can you give me an example?

**Comment by diizzyy** [comment]
That looks really nice!

I guess this is kinda in-ish topic but since you're doing nginx stuff andgreat work on the luci related stuff too! :-)

Could you have a look at getting these packaged?
https://github.com/arut/nginx-rtmp-module
https://github.com/arut/nginx-ts-module

**Comment by Ansuel** [comment]
@diizzyy will include them... just one question...

nginx-rtmp-module require ssl module or not? 

In the index page is not that clear...
`Several versions of nginx (1.3.14 - 1.5.0) require http_ssl_module to be added as well:`

**Comment by diizzyy** [comment]
@Ansuel 
It requires OpenSSL

**Comment by Ansuel** [comment]
Then i will depend that config on that...

**Comment by Ansuel** [comment]
Any update about this? @hnyman

**Comment by hnyman** [comment]
Would be great if @heil would look at the commit.

And @Ansuel , you might maybe add yourself as the second maintainer to the package, as you seem to be tweaking it quite a lot for LuCI compatibility.

**Comment by Ansuel** [comment]
@hnyman added as secondary maintainer

**Comment by Ansuel** [comment]
any update on this? @hnyman

**Comment by hnyman** [comment]
> Do you know any info about the problem @karlp notice?

No detailed info about that. Sorry.

**Comment by Ansuel** [comment]
Could be the fact that the virtual ackage have the same name of the variant as this could create something like nginx-no-ssl
nginx-ssl-ssl nginx-all-modules-all-modules
Will check what is created...

**Comment by Ansuel** [comment]
looking at the code produced, all looks good...

It's 4 weeks that i build firmware with this commit and i can't see any problem...

**Comment by Ansuel** [comment]
the build/configure was causing the recompile problem... fixed.

**Comment by hnyman** [comment]
I will merge this, and we can then see the fallout.

**Pull request merged**